### PR TITLE
[Transform] prevent assignment to nodes older than 7.4

### LINKED ISSUE #48019: [Transform] transform jobs on nodes < 7.4.0 mixed with nodes > 7.4 might fail
Nodes in all versions of 7.2 and 7.3 might not see transform documents created in 7.4, 7.5 ...

Only _after_ the switch to a versioned index in #45375 (7.4.0) retrieval uses a search pattern to read back documents (config, checkpoint, state). This allows to retrieve jobs even if the cluster is partially upgraded. But if nodes are on version 7.3 and 7.4 the hardcoded index name those documents might not be retrievable.

Long story short: Mixed version clusters with nodes <7.4 together with nodes >=7.4 do not work reliable. 

Solution: 

 - disable node allocation if any node < 7.4. This will prevent placing jobs on mixed clusters until the upgrade has completed
 - document as known limitation: transforms do not execute during a rolling upgrade from 7.2/7.3 to 7.x

### PR DESCRIPTION
**Pull request by hendrikmuhs**
do not allow assignment of transforms to nodes older than 7.4 (in mixed clusters).

fixes #48019

### CODE REVIEW

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/2

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/bwc

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java [hunk 1]
```diff
@@ -100,7 +101,10 @@ public TransformPersistentTasksExecutor(Client client,
             return new PersistentTasksCustomMetaData.Assignment(null, reason);
         }
         DiscoveryNode discoveryNode = selectLeastLoadedNode(clusterState, (node) ->
-            node.isDataNode() && node.getVersion().onOrAfter(params.getVersion())
+            node.isDataNode() &&
+            // see gh#48019 older nodes might not be able to read documents
+            node.getVersion().onOrAfter(Version.V_7_4_0) &&
+            node.getVersion().onOrAfter(params.getVersion())
```
Comment (benwtrent) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java]: I think we should not allow assignment at all in a mixed cluster where a `<= 7.3` node is present. If the task is running and somebody makes a stats call, that stats call could be made against a 7.3 node which queries the wrong index and incorrectly says that the transform does not exist.

**Comment by hendrikmuhs** [comment]
superseeded by #48055

**Pull request closed without merging**